src/scripts/migrate.py

```diff
@@ -0,0 +1,143 @@
+"""Slurp migration pipeline
+
+Basically, we:
+1. Look at all unmapped terms T
+2. If all of parents of T are mapped, designate for slurping (we only slurp if parents are already slurped, iteratively)
+3. Extract basic information about T and export as ROBOT template
+
+Resources
+- https://incatools.github.io/ontology-access-kit/
+- https://incatools.github.io/ontology-access-kit/intro/tutorial02.html
+
+TODO's:
+  -
+"""
+import os
+import pickle
+from argparse import ArgumentParser
+from typing import Dict, List, Set, Union
+
+import pandas as pd
+import yaml
+from oaklib.implementations import ProntoImplementation
+from oaklib.resource import OntologyResource
+from oaklib.types import CURIE, URI
+
+from migrate_utils import CACHE_DIR, PREFIX, Term, _get_all_owned_terms, _get_next_available_mondo_id, \
+    get_mondo_term_ids
+
+
+def _load_ontology(ontology_path: str, use_cache=False) -> ProntoImplementation:
+    """Load ontology"""
+    cache_path_ontology = os.path.join(
+        CACHE_DIR, f'migrate_{os.path.basename(ontology_path).replace(".owl", "")}.pickle')
+
+    if use_cache and os.path.exists(cache_path_ontology):
+        # todo: Does loading from cache even save time? seemed to take ~16 sec
+        ontology = pickle.load(open(cache_path_ontology, 'rb'))
+    else:
+        # todo's: Ontology implementation selection
+        #  i. If trouble, can try `SparqlImplementation`, but ~6 min to load and queries slow (cuz rdflib)
+        #  ii. use Sql > Pronto when .entities() fixed: https://github.com/INCATools/ontology-access-kit/issues/235
+        # ontology = SqlImplementation(OntologyResource(slug=ontology_path, local=True))
+        ontology = ProntoImplementation(OntologyResource(slug=ontology_path, local=True))  # ~17 sec
+        if use_cache:
+            with open(cache_path_ontology, 'wb') as f:
+                pickle.dump(ontology, f, protocol=pickle.HIGHEST_PROTOCOL)
+
+    return ontology
+
+
+def slurp(
+    ontology_path: str, onto_config_path: str, sssom_map_path: str, min_id: int, mondo_terms_path: str, outpath: str,
+    use_cache=False
+) -> pd.DataFrame:
+    """Run slurp pipeline for given ontology"""
+    # Read inputs
+    ontology: ProntoImplementation = _load_ontology(ontology_path, use_cache)
+    with open(onto_config_path, 'r') as stream:
+        onto_config = yaml.safe_load(stream)
+        owned_prefix_map: Dict[PREFIX, URI] = onto_config['base_prefix_map']
+    sssom_df: pd.DataFrame = pd.read_csv(sssom_map_path, comment='#', sep='\t')
+    mondo_term_ids: Set[int] = get_mondo_term_ids(mondo_terms_path)
+
+    # Intermediates
+    owned_terms: List[Term] = _get_all_owned_terms(
+        ontology=ontology, owned_prefix_map=owned_prefix_map, cache_dir_path=CACHE_DIR,
+        onto_config_path=onto_config_path)
+    sssom_object_ids: Set[Union[URI, CURIE]] = set(sssom_df['object_id'])  # Usually CURIE, but spec allows URI
```

Comment:
FYI: `sssom_object_ids`: I set the type to `Set[Union[URI, CURIE]]` and allowed the code to handle this.  it would be easier if just CURIE only, but the SSSOM spec says that the type is `RDF ref`, which i think allows for either a CURIE or URI.